### PR TITLE
Fix fail to prepare tablet reader bug

### LINKED ISSUE #5153: query failed because fail to prepare tablet reader
### Steps to reproduce the behavior (Required)
when one replica's publish version stuck in BE, FE will set the wrong visible version for replica, which leads to the failure of preparing tablet reader.
The following is the step to reproduce:
1. make publish version stuck in BE for one minute(maybe add sleep in publish version for test)
2. insert into one table twice
3. select * from table

then you will come into failure of "fail to prepare tablet reader"

### Expected behavior (Required)

query success

### Real behavior (Required)
fail to prepare tablet reader

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
5.0.3

### PR DESCRIPTION
**Pull request by ABingHuang**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] others

## Which issues of this PR fixes ：

Fixes #5153 

## Problem Summary(Required) ：

When publish stuck in one replica and the last failed version is > 0,
the replica will not be treated as an error replica, and FE will update
the replica's version to the newest visible version of partition when the replica
report. and the query will fail because missing version because the publish
the version does not finish in BE; the version is not queryable.
And previously, the quorum success of the published version depended on the tablet report. 
I added an optimization to decide quorum success based on the publishing version task.
It will boost the publish version process from 10s+ to 0.03s.

### CODE REVIEW

**Review by chaoyli — DISMISSED** [review]

**Review by imay — DISMISSED** [review]
LGTM

File: fe/fe-core/src/main/java/com/starrocks/transaction/DatabaseTransactionMgr.java [hunk 1]
```diff
@@ -690,7 +690,7 @@ public boolean canTxnFinished(TransactionState txn, Set<Long> errReplicas, Set<L
                             // if most replica's version have been updated to version published
                             // which means publish version task finished in replica  
                             for (Replica replica : ((LocalTablet) tablet).getReplicas()) {
-                                if (!errReplicas.contains(replica.getId()) && replica.getLastFailedVersion() < 0) {
```
Comment (imay) [inline fe/fe-core/src/main/java/com/starrocks/transaction/DatabaseTransactionMgr.java:693]: When `replica.getLastFailedVersion() < 0` is false, healthReplicaNum won't be added, and it won't be greater than quorumNum. And this function will return false.
Your changing will only make healthReplicaNum become greater, which can not resolve the problem in your PR description.

Comment (ABingHuang) [reply fe/fe-core/src/main/java/com/starrocks/transaction/DatabaseTransactionMgr.java:693]: it is the problem of errorReplica. I keep the condition getLastFailedVersion < 0 for healthyNum counter and remove it for errorReplicas.

**Comment by ABingHuang** [comment]
previously, the quorum success of publish version depends on the tablet report. I add a optimization to decide quorum success based on publish version task. It will boost the publish version process, from 10 seconds+ to 0.03s.

**Review by wyb — DISMISSED** [review]

**Review by wyb — APPROVED** [review]

**Review by imay — APPROVED** [review]
LGTM

**Pull request merged**